services/utils.ts

```diff
@@ -233,46 +262,50 @@ export async function completeComponent(comp, courseId, isComplete) {
   StorageService.updateStudentInfo(updatedStudent);
 
   return { points, updatedStudent };
-}
+};
 
-export function isCourseCompleted(student) {
-  return student.courses.some(
-    (course) => checkProgressCourse(course.Id) === 100,
+const isCourseCompleted = (student: StudentProgress): boolean => {
+  // A course is considered completed if all components in all sections are complete
+  return student.courses.some((course) =>
+    course.sections.every((section) =>
+      section.components.every((component) => component.isComplete === true),
+    ),
   );
-}
+};
 
-export function isSectionCompleted(student, sectionId) {
+const isSectionCompleted = (student: StudentProgress, sectionId: string) => {
   return student.courses.some((course) =>
-    course.sections.some(
-      (section) => section.sectionId == sectionId && section.isComplete,
-    ),
+    course.sections.some((section) => section.sectionId === sectionId),
   );
-}
+};
 
-export function isComponentCompleted(student, compId) {
+const isComponentCompleted = (student: StudentProgress, compId: string) => {
   return student.courses.some((course) =>
     course.sections.some((section) =>
       section.components.some(
         (component) => component.compId === compId && component.isComplete,
       ),
     ),
   );
-}
+};
 
-function isFirstAttemptExercise(student, compId) {
+const isFirstAttemptExercise = (student: StudentProgress, compId: string) => {
   return student.courses.some((course) =>
     course.sections.some((section) =>
       section.components.some(
         (component) => component.compId === compId && component.isFirstAttempt,
       ),
     ),
   );
-}
+};
 
-// Returns the students progress of a course in percentage
-export async function checkProgressCourse(courseId) {
+// Returns the students progress of a course in percentage, and also returns completed and total components
+const checkProgressCourse = async (
+  courseId: string,
+): Promise<[number, number, number]> => {
```

Comment:
Isn't this function returning `ProgressTuple` from https://github.com/ErasmusEgalitarian/educado-mobile/pull/456/files#diff-e4511dbf610981193d68ec14a0f14a7b62565c98b69a2db41ea6be8c2adbea46R18? If yes, `ProgressTuple` should be a global type exported from `types/` (inline, named export) and be imported in this file and `components/Courses/CourseCard/CourseCard.tsx`

```suggestion
): Promise<ProgressTuple> => {
```